pyonfx/timestamps.py

```diff
@@ -0,0 +1,552 @@
+                if lowest_timestamp is None or lowest_timestamp > timestamp:
+                    lowest_timestamp = timestamp
+                timestamps.append(rounding_method(timestamp))
+
+            timestamps.sort()
+            return lowest_timestamp, highest_timestamp, timestamps
+
+        # Verify if ffprobe is installed
+        if shutil.which("ffprobe") is None:
+            raise Exception("ffprobe is not in the environment variable.")
+
+        if not os.path.isfile(video_path):
+            raise FileNotFoundError(f'Invalid path for the video file: "{video_path}"')
+
+        # Getting video absolute path and checking for its existance
+        if not os.path.isabs(video_path):
+            dirname = os.path.dirname(os.path.abspath(sys.argv[0]))
+            video_path = os.path.join(dirname, video_path)
+
+        cmd = [
+            "ffprobe",
+            "-select_streams",
+            f"{index}",
+            "-show_entries",
+            "packet=pts_time,dts_time:stream=codec_type,time_base:format=format_name",
+            f"{video_path}",
+            "-print_format",
+            "json",
+        ]
+        ffprobe_output = subprocess.run(cmd, capture_output=True, text=True)
+        ffprobe_output_dict = json.loads(ffprobe_output.stdout)
+
+        if not ffprobe_output_dict:
+            raise ValueError(f"The file {video_path} is not a video file.")
+
+        if len(ffprobe_output_dict["streams"]) == 0:
+            raise ValueError(f"The index {index} is not in the file {video_path}.")
+
+        if ffprobe_output_dict["streams"][0]["codec_type"] != "video":
+            raise ValueError(
+                f'The index {index} is not a video stream. It is an "{ffprobe_output_dict["streams"][0]["codec_type"]}" stream.'
+            )
+
+        if ffprobe_output_dict["format"]["format_name"] == "matroska,webm":
+            # We only do this check for .mkv file. See the note about mkv in the class documentation
+            time_base = Fraction(ffprobe_output_dict["streams"][0]["time_base"])
+            # 1/1000 represent 1 ms. If the time_base cannot divided by 1/1000, then it means that the timestamps aren't rounded
+            if time_base % (1 / 1000):
+                warnings.warn(
+                    "Your mkv file isn't perfectly rounded to ms. In this situation, you may prefer to use RoundingMethod.floor then RoundingMethod.ROUND.",
+                    UserWarning,
+                )
+
+        first_frame_time, last_frame_time, timestamps = get_timestamps(
+            ffprobe_output_dict["packets"]
+        )
+        fpms = Fraction(len(timestamps) - 1) / (last_frame_time - first_frame_time)
+
+        return cls(
+            rounding_method=rounding_method,
+            timestamps=timestamps,
+            normalize=normalize,
+            fpms=fpms,
+            last_frame_time=last_frame_time,
+        )
+
+    @staticmethod
+    def validate(timestamps: List[int]) -> None:
```

Comment:
Since it doesn't do anything fancy, it seems to be called only in the `init` method and seems like the only place where it should be called, I would rather remove it and incorporate the 3 checks in the `init` to reduce the codebase.

Reply:
Done